### PR TITLE
refactor(identify): simplify reply handling

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Description

A `ConnectionHandler` is bound to a single peer. There is no need to embed the `PeerId` in the event that we report to the behaviour, it already knows which peer the event relates to.

## Notes & open questions

~Draft because it is a stacked PR.~

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: swarm/CHANGELOG.md [hunk 1]
```diff
@@ -42,6 +42,14 @@
   
   See [PR 3651].
 
+- Allow `ConnectionHandler`s to report and learn about the supported protocols on a connection.
+  The newly introduced API elements are:
+  - `ConnectionHandlerEvent::ReportRemoteProtocols`
+  - `ConnectionEvent::LocalProtocolsChange`
+  - `ConnectionEvent::RemoteProtocolsChange`
+  
+  See [PR 3651].
```
Comment (mxinden) [inline swarm/CHANGELOG.md]: Bad merge?

**Review by mxinden — COMMENTED** [review]
Are the 63 commits intended?

File: swarm/CHANGELOG.md (hunk 1 above)
Comment (thomaseizinger) [reply swarm/CHANGELOG.md]: Yes!

File: swarm/CHANGELOG.md [hunk 2]
```diff
@@ -42,6 +42,14 @@
...
+
```
Comment (thomaseizinger) [inline swarm/CHANGELOG.md]:
```suggestion
```

**Comment by thomaseizinger** [comment]
> Are the 63 commits intended?

It is because this was stacked on top of another PR. Shouldn't matter when we squash, right?

**Review by mxinden — APPROVED** [review]

**Pull request merged**